### PR TITLE
Improvements to DocumentDB 

### PR DESCRIPTION
**Pull request by brettferdosi**
Made a few improvements to the DocumentDB typing:
- Added support for SQL parameterization (see http://azure.microsoft.com/blog/2015/01/14/announcing-sql-parameterization-in-documentdb/)
- Used union types to update all query methods to accept either query strings or parameterized queries
- Added support for replacing documents and creating/querying triggers
- Added appropriate tests to the test file

### CODE REVIEW

**Comment by vvakame** [comment]
*documentdb/documentdb.d.ts*

to author(@NoelAbrahams). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

File: documentdb/documentdb.d.ts [hunk 1]
```diff
@@ -52,7 +52,25 @@ declare module 'documentdb' {
...
     }
-
+    
+    /** The Sql query parameter. */
+    interface SqlParameter {
+        /** The name of the parameter. */
+        name: string;
+        
+        /** The value of the parameter. */
+        value: string;
```
Comment (NoelAbrahams) [inline documentdb/documentdb.d.ts]: The spec says 

> Parameter values can be any valid JSON (strings, numbers, Booleans, null, even arrays or nested JSON)

so we should probably leave `value` as `any`, no?

**Comment by brettferdosi** [comment]
Thanks for the catch. I did see that in the article, but left `value` as a string because the documentdb SDK defines SqlParameter like this in documentdb/lib/documentclient.js:

```
/**
  * The Sql query parameter.
  * @typedef {Object} SqlParameter
  * @property {string} name         -       The name of the parameter.
  * @property {string} value        -       The value of the parameter.
  */
```

where `value` is a string. But I just did some tests with `value` as a number of different types on a real database, and it seems that value can be `any`--maybe the SDK just hasn't been updated to reflect that.

I just published that change and added a test case/made some fixes to the test file. Please let me know if they look good.

Brett

**Comment by NoelAbrahams** [comment]
Looks good, thanks. :+1:

**Comment by vvakame** [comment]
@NoelAbrahams thanks!
@brettferdosi thanks mate!

**Pull request merged**